src/plugins/main_menu.rs

```diff
@@ -1 +1,19 @@
-//use bevy::prelude::*;
+use bevy::prelude::*;
+
+use crate::{
+    visuals::main_menu::*,
+    mechanics::main_menu_buttons::*,
+    AppState,
+};
+
+pub struct MainMenuPlugin;
+
+impl Plugin for MainMenuPlugin {
+    fn build(&self, app: &mut App) {
+        app
+        .add_system(spawn_main_menu.in_schedule(OnEnter(AppState::MainMenu))            )
+        .add_systems((load_background_image,load_button_image,load_text_font))      
+        .add_system(button_system)
```

Comment:
Normally I would bind these to OnUpdate:
 ```rust
.add_systems((load_background_image, load_button_image, load_text_font)
    .in_set(OnUpdate(AppState::MainMenu)))
.add_system(button_system.in_set(OnUpdate(AppState::MainMenu)))
```
but looking at the code for each of these systems, since a lot of it depends upon these elements existing or being added in the first place, this suggestion would more be a safety precaution.

Reply:
Good idea